### PR TITLE
Multi-arch Docker image support

### PR DESCRIPTION
**Pull request by ilteoood**
With this PR, I've introduced the following changes:
- Migration from circle-ci to github actions
- Minor and patch dependencies upgrade
- Introduction of multi-arch docker image

### CODE REVIEW

**Comment by davidebianchi** [comment]
Hi @ilteoood, thanks for the PR! 
How do you think this docker image should be used?

**Comment by ilteoood** [comment]
ping @davidebianchi

**Comment by ilteoood** [comment]
ping @davidebianchi

**Comment by ilteoood** [comment]
ping @davidebianchi

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: mock-server CI
+
+on:
+  push:
+    branches: [ master, main ]
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:5]:
```suggestion
    branches: [ master ]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    tags: '*'
+  pull_request:
+    branches: [ master, main ]
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:8]:
```suggestion
    branches: [ master ]
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Login to npm
+        run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
+      - name: Publish to npm
+        run: npm publish
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:46]:
```suggestion
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
      - name: Install dependencies
        run: yarn install --frozen-lockfile
      - run: npm publish
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  docker-build:
+    name: Build docker
+    needs: qa
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main' }}
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:52]:
```suggestion
    if: ${{ startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master' }}
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Prepare
+        id: prep
+        run: |
+          DOCKER_IMAGE=@staticdeploy/mock-server
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:59]: It should be better to use github registry to upload this image

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,23 @@
+FROM node:gallium-alpine
+RUN apk add tini --no-cache
+
+LABEL name="@staticdeploy/mock-server" \
+      description="Easy to use, no frills mock server" \
+      io.staticdeploy.url="https://staticdeploy.io/" \
+      io.staticdeploy.version="3.0.0"
+
+ENV PORT=3456
+ENV ROOT=mock-server
```
Comment (davidebianchi) [inline Dockerfile:10]:
```suggestion
ENV ROOT=mock-server
ENV NODE_ENV=production
```

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+COPY package.json .
+COPY yarn.lock .
+
+RUN yarn install --frozen-lockfile
+
+COPY src ./src
+
+USER node
+
+CMD /sbin/tini ./src/bin/index.js --port=${PORT} --root=${ROOT}
```
Comment (davidebianchi) [inline Dockerfile:23]:
```suggestion
ENTRYPOINT ["/sbin/tini", "--"]

CMD ./src/bin/index.js --port=${PORT} --root=${ROOT}
```

**Review by davidebianchi — CHANGES_REQUESTED** [review]
Thank you for the PR, and sorry for the late review!

File: .github/workflows/ci.yml (hunk 5 above)
Comment (davidebianchi) [reply .github/workflows/ci.yml:62]:
```suggestion
          DOCKER_IMAGE=staticdeploy/mock-server
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+          echo ::set-output name=image::${DOCKER_IMAGE}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
```
Comment (davidebianchi) [inline .github/workflows/ci.yml:81]: This should be a login to github registry, so something like:

```suggestion
      - name: Docker Login to ghcr.io
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GITHUB_TOKEN }}

```

File: .github/workflows/ci.yml (hunk 5 above)
Comment (ilteoood) [reply .github/workflows/ci.yml:62]:
```suggestion
          DOCKER_IMAGE=staticdeploy/mock-server
```

**Review by davidebianchi — APPROVED** [review]
LGTM

**Pull request merged**